MiniApp/Classes/core/Storage/MiniAppKeyChain.swift

```diff
@@ -74,11 +76,78 @@ import Foundation
         return supportedPermissionList
     }
 
-    private func write(keys: KeysDictionary) {
+    private func writePermissionInfo(keys: KeysDictionary) {
         guard let data = try? JSONEncoder().encode(keys) else {
             return
         }
+        setValueInKeyChain(data: data)
+    }
+
+    private func retrieveAllPermissions() -> KeysDictionary? {
+        guard let storedData = getAllKeysFromKeychain() else {
+            return nil
+        }
+
+        guard let keys = ResponseDecoder.decode(decodeType: KeysDictionary.self, data: storedData) else {
+            return nil
+        }
+
+        return keys
+    }
+
+    // MARK: - Cache Verifier Methods
+    internal func setCacheInfo(key: String, for keyId: String) {
```

Comment:
I think the naming of these methods isn't very clear. `Cache` could refer to many different things that we're caching in the SDK - we're also caching permissions in this same class. Maybe we could rename it with `CacheVerification`? 
- `setCacheVerificationInfo`
- `getCacheVerificationInfo`
- etc...

Reply:
Done